chart/templates/networkpolicy.yaml

```diff
@@ -1,108 +1,251 @@
 {{- if .Values.policies.networkPolicy.enabled }}
+{{- $labels := merge dict (dict
+      "app" "vcluster"
+      "chart" (include "vcluster.version.label" $)
+      "release" .Release.Name
+      "heritage" .Release.Service) .Values.policies.networkPolicy.labels -}}
+{{- $annotations := merge dict .Values.controlPlane.advanced.globalMetadata.annotations .Values.policies.networkPolicy.annotations -}}
+{{- $labelsAndAnnotations := dict "labels" $labels "annotations" $annotations -}}
+---
 apiVersion: networking.k8s.io/v1
 kind: NetworkPolicy
 metadata:
   name: vc-work-{{ .Release.Name }}
   namespace: {{ .Release.Namespace }}
-  labels:
-    app: vcluster
-    chart: "{{ include "vcluster.version.label"  $ }}"
-    release: "{{ .Release.Name }}"
-    heritage: "{{ .Release.Service }}"
-    {{- if .Values.policies.networkPolicy.labels }}
-{{ toYaml .Values.policies.networkPolicy.labels | indent 4 }}
-    {{- end }}
-  {{- $annotations := merge dict .Values.controlPlane.advanced.globalMetadata.annotations .Values.policies.networkPolicy.annotations }}
-  {{- if $annotations }}
-  annotations:
-{{ toYaml $annotations | indent 4 }}
-  {{- end }}
+{{ toYaml $labelsAndAnnotations | indent 2 }}
 spec:
   podSelector:
     matchLabels:
       vcluster.loft.sh/managed-by: {{ .Release.Name }}
+  policyTypes:
+    - Egress
+    - Ingress
   egress:
-    # Allows outgoing connections to the vcluster control plane
+    # Allow egress to vcluster DNS and control plane.
```

Comment:
So as far as I understood we do the following:

Egress:
* Allow dns / control plane
* Allow workloads
* Allow public egress

Ingress:
* Allow control plane
* Allow workloads

Why do we not need dns and public for ingress?